pandas/tests/reshape/merge/test_multi.py

```diff
@@ -460,6 +460,33 @@ def test_merge_na_keys(self):
 
         tm.assert_frame_equal(result, expected)
 
+    def test_merge_datetime_index_empty_df(self):
+
+        date = np.array(
+            [pd.Timestamp("1950-01-01"), pd.Timestamp("1950-01-02")],
```

Comment:
don't specify dtypes here and you don't need to use np.array (list are fine)

Reply:
Do you mean this comment to apply to https://github.com/pandas-dev/pandas/pull/36897/files/63312ce177f826c67a92b3f7122718bc07854254#diff-4a53bad084aad7ae351043e2cd13bf62R479 as well?
The merge operator will overwrite any assigned data type with "object" and so I explicitly construct an array of NaN with an object data-type so the comparison will pass.

Reply:
bump on this

Reply:
@PCerles pls call this `dates` or `datecol` instead of `date`, dont put np.array, just `datecol = [pd.Timestamp("1950-01-01"), pd.Timestamp("1950-01-02")]`